pkg/controller/storagecluster/cephobjectstores.go

```diff
@@ -26,16 +26,14 @@ func (r *ReconcileStorageCluster) ensureCephObjectStores(instance *ocsv1.Storage
 		return nil
 	}
 
-	cephObjectStores, err := r.newCephObjectStoreInstances(instance)
+	var cephObjectStores []*cephv1.CephObjectStore
+	if cephObjectStores, err = r.newCephObjectStoreInstances(instance); err == nil {
+		err = r.createCephObjectStores(cephObjectStores, instance, reqLogger)
+	}
 	if err != nil {
+		reqLogger.Error(err, "could not create CephObjectStores")
```

Comment:
These changes don't seem related to the commit message. This can probably be integrated into the next commit.

Reply:
Moved to a separate commit...